fabtests/examples/example_rdm.c

```diff
@@ -0,0 +1,507 @@
+/*
+ * Copyright (c) Intel Corporation.  All rights reserved.
+ *
+ * This software is available to you under the BSD license
+ * below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <rdma/fabric.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_endpoint.h>
+
+//Run Server: example_rdm
+//Run client: example_rdm <server_ip>
+
+#define BUF_SIZE 64
+
+char *dst_addr = NULL;
+char *port = "9228";
+struct fi_info *hints, *info;
+struct fid_fabric *fabric = NULL;
+struct fid_domain *domain = NULL;
+struct fid_ep *ep = NULL;
+struct fid_av *av = NULL;
+struct fid_cq *cq = NULL;
+char buf[BUF_SIZE];
+fi_addr_t fi_addr = FI_ADDR_UNSPEC;
+
+/*
+ * Initializes all basic OFI resources to allow for a server/client to
+ * exchange a message
+ */
+static int initialize(void)
+{
+	struct fi_cq_attr cq_attr = {0};
+	struct fi_av_attr av_attr = {0};
+	const struct sockaddr_in *sin;
+	char str_addr[INET_ADDRSTRLEN];
+	int ret;
+
+	/*
+	 * The first OFI call to happen for initialization is fi_getinfo which
```

Comment:
The first call is usually fi_allocinfo() for the hints.  See comment below about moving hints initialization above this function.